### ISSUE TITLE
conda skeleton cpan --recursive Test::Deep

### ISSUE THREAD

**Issue by jerowe** [conda/conda-build#994]
Error:

```
Using url https://cpan.metacpan.org/authors/id/R/RJ/RJBS/Test-Deep-1.120.tar.gz (38973) for Test-Deep.
Processing dependencies for Test-Deep.....Using Anaconda Cloud api site https://api.anaconda.org
Fetching package metadata: ....
.An unexpected error has occurred, please consider sending the
following traceback to the conda GitHub issue tracker at:


https://github.com/conda/conda-build/issues


Include the output of the command 'conda info' in your report.

Traceback (most recent call last):
  File "/home/jillian/.local/easybuild/software/anaconda/2-4.0.0/bin/conda-skeleton", line 5, in <module>
    sys.exit(main())
  File "/home/jillian/.local/easybuild/software/anaconda/2-4.0.0/lib/python2.7/site-packages/conda_build/main_skeleton.py", line 298, in main
    args_func(args, p)
  File "/home/jillian/.local/easybuild/software/anaconda/2-4.0.0/lib/python2.7/site-packages/conda_build/main_build.py", line 493, in args_func
    args.func(args, p)
  File "/home/jillian/.local/easybuild/software/anaconda/2-4.0.0/lib/python2.7/site-packages/conda_build/main_skeleton.py", line 315, in execute
    cpan.main(args, parser)
  File "/home/jillian/.local/easybuild/software/anaconda/2-4.0.0/lib/python2.7/site-packages/conda_build/cpan.py", line 249, in main
    processed_packages)
  File "/home/jillian/.local/easybuild/software/anaconda/2-4.0.0/lib/python2.7/site-packages/conda_build/cpan.py", line 407, in deps_for_package
    dep_version, perl_version, dependency=True)
  File "/home/jillian/.local/easybuild/software/anaconda/2-4.0.0/lib/python2.7/site-packages/conda_build/cpan.py", line 539, in get_release_info
    core_version = core_module_version(orig_package, perl_version)
  File "/home/jillian/.local/easybuild/software/anaconda/2-4.0.0/lib/python2.7/site-packages/conda/utils.py", line 118, in **call**
    value = self.func(_args, *_kw)
  File "/home/jillian/.local/easybuild/software/anaconda/2-4.0.0/lib/python2.7/site-packages/conda_build/cpan.py", line 340, in core_module_version
    if LooseVersion(first_version) > LooseVersion(version):
  File "/home/jillian/.local/easybuild/software/anaconda/2-4.0.0/lib/python2.7/distutils/version.py", line 265, in **init**
    self.parse(vstring)
  File "/home/jillian/.local/easybuild/software/anaconda/2-4.0.0/lib/python2.7/distutils/version.py", line 274, in parse
    self.component_re.split(vstring))
TypeError: expected string or buffer
`
Output of conda info

`
Current conda install:

```
         platform : linux-64
    conda version : 4.0.5
```

  conda-build version : 1.20.0
       python version : 2.7.11.final.0
     requests version : 2.9.1
     root environment : /home/jillian/.local/easybuild/software/anaconda/2-4.0.0  (writable)
  default environment : /home/jillian/.local/easybuild/software/anaconda/2-4.0.0
     envs directories : /home/jillian/.local/easybuild/software/anaconda/2-4.0.0/envs
        package cache : /home/jillian/.local/easybuild/software/anaconda/2-4.0.0/pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/linux-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : /home/jillian/.condarc
    is foreign system : False
```

It also fails not using the recursive flag.

**Comment by msarahan** [comment]
The 2.0 release will have a pretty significant rework of these skeleton generators.  If I forget, please remind me to revisit this issue once 2.0 is out.

**Comment by jerowe** [comment]
@msarahan , I am still experiencing this with the new cpan skeleton.

One thing that I have noticed is that the new cpan api takes care of a lot of the issues the skeleton is checking for.

'lib', for instance, is a core perl module. Previously, it would not have been under the 'releases' api, but only the module. Now, as far as I can tell, all the perl libraries are under the 'release' endpoint, and core modules are also under 'module'.

How do you feel about me digging into the cpan skeleton more to possibly update it against the new cpan api? I use perl enough that I actually need this, so either I build it here or I will need to roll my own solution.

**Comment by msarahan** [comment]
@jerowe please don't be shy.  Any fixes/improvements you make to cpan or the api in general are most welcome.  It's fine even if you end up needing to break the API - it will just take longer to roll out, since we'll need some kind of deprecation cycle.

Anything not in the conda_build.api module is totally open to change any time, as long as the api remains functionally equivalent.  For cpan, this means as long as it is spitting out the right recipes, I'm quite happy no matter how much the code changes.

**Comment by jerowe** [comment]
HAHAHA FREEDOM!

Like I said earlier, I use develop in perl quite a bit, so this will be very useful for me to have. ;-) I need this for some other projects I need to wrap up, so I should be able to get working on it soon.

**Comment by timsnyder** [comment]
Looks like this may have been resolved by PR #2156.  I'm able to run `conda skeleton cpan --recursive Test::Deep` today without running into errors.
